### PR TITLE
Added conditional to only collect qmpstatus on qemu VMs

### PR DESCRIPTION
**Pull request by Thulium-Drake**
##### SUMMARY

This fixes an issue with LXC containers after #4723 was implemented. I completely missed this when reviewing it and the unit tests didn't cover it :-)

I'm not too sure however on how to update the unit tests in order to fix this (yet, but it's late, unless I come up with an idea earlier then someone else, I'll adjust the tests as well :-) )

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

proxmox inventory
##### ADDITIONAL INFORMATION

Running the module without this task results in the errors below:

```paste below
[WARNING]:  * Failed to parse /opt/ansible/projects/element-networks/inventory/04_nc_proxmox.yml with
ansible_collections.community.general.plugins.inventory.proxmox plugin: 'qmpstatus'
  File "/usr/lib/python3/dist-packages/ansible/inventory/manager.py", line 290, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/opt/ansible/projects/element-networks/collections/ansible_collections/community/general/plugins/inventory/proxmox.py", line 656, in parse
    self._populate()
  File "/opt/ansible/projects/element-networks/collections/ansible_collections/community/general/plugins/inventory/proxmox.py", line 597, in _populate
    name = self._handle_item(node['node'], ittype, item)
  File "/opt/ansible/projects/element-networks/collections/ansible_collections/community/general/plugins/inventory/proxmox.py", line 507, in _handle_item
    self._get_vm_status(properties, node, vmid, ittype, name)
  File "/opt/ansible/projects/element-networks/collections/ansible_collections/community/general/plugins/inventory/proxmox.py", line 446, in _get_vm_status
    properties[self._fact('qmpstatus')] = ret['qmpstatus']

```

### CODE REVIEW

**Comment by ilijamt** [comment]
@Thulium-Drake do you think maybe we can just use a default value if not set instead of a vm type check. As `qmpstatus` doesn't mean it's set even if you have QEMU machine, it will only be set when there are facts. 

Yeah, weird that the tests missed it.

**Comment by ilijamt** [comment]
The tests missed the issue because we mock the `_get_json`. 

https://github.com/ansible-collections/community.general/blob/b556b142ecdb697cf729e902aeafdeda8dff2f92/tests/unit/plugins/inventory/test_proxmox.py#L570

This will require a bit more work to add tests for this.

**Review by felixfontein — COMMENTED** [review]
Thanks for the fix! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

File: plugins/inventory/proxmox.py [hunk 1]
```diff
@@ -443,7 +443,8 @@ def _get_vm_config(self, properties, node, vmid, vmtype, name):
     def _get_vm_status(self, properties, node, vmid, vmtype, name):
         ret = self._get_json("%s/api2/json/nodes/%s/%s/%s/status/current" % (self.proxmox_url, node, vmtype, vmid))
         properties[self._fact('status')] = ret['status']
-        properties[self._fact('qmpstatus')] = ret['qmpstatus']
+        if vmtype == 'qemu':
```
Comment (ilijamt) [inline plugins/inventory/proxmox.py:446]: @Thulium-Drake tnx for fixing this 

We should change this line to this. This is only applicable when we have all 3 options set.  Like on this line 
https://github.com/ansible-collections/community.general/blob/59c6fad7c728bb7cae93c39ecb2aaa813ca4974b/plugins/inventory/proxmox.py#L524

```
if self.get_option('want_facts') and vmtype == 'qemu' and self.get_option('qemu_extended_statuses'):
```

Comment (Thulium-Drake) [reply plugins/inventory/proxmox.py:446]: Sounds like a good plan, fixed!

**Comment by Thulium-Drake** [comment]
For some reason I run into some weird errors when trying the unit tests on my laptop, but I made some changes to them. It should now 'just' collect statuses from the mocked API instead of noop-ing :-)

**Comment by Thulium-Drake** [comment]
That should about cover it! @ilijamt can you have a look-see?

File: plugins/inventory/proxmox.py [hunk 2]
```diff
@@ -443,7 +443,8 @@ def _get_vm_config(self, properties, node, vmid, vmtype, name):
     def _get_vm_status(self, properties, node, vmid, vmtype, name):
         ret = self._get_json("%s/api2/json/nodes/%s/%s/%s/status/current" % (self.proxmox_url, node, vmtype, vmid))
         properties[self._fact('status')] = ret['status']
-        properties[self._fact('qmpstatus')] = ret['qmpstatus']
+        if self.get_option('want_facts') and vmtype == 'qemu':
```
Comment (felixfontein) [inline plugins/inventory/proxmox.py:446]: This function is only called when `self.get_option('want_facts')`. Thus it doesn't make sense to check that here again.
```suggestion
        if vmtype == 'qemu':
```

Comment (Thulium-Drake) [reply plugins/inventory/proxmox.py:446]: Oh, yeah, right... :sweat_smile:

File: changelogs/fragments/4816-proxmox-fix-extended-status.yaml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - proxmox inventory - fixed extended status detection for qemu (https://github.com/ansible-collections/community.general/pull/4816).
```
Comment (felixfontein) [inline changelogs/fragments/4816-proxmox-fix-extended-status.yaml:2]:
```suggestion
  - proxmox inventory plugin - fixed extended status detection for qemu (https://github.com/ansible-collections/community.general/pull/4816).
```

Comment (Thulium-Drake) [reply changelogs/fragments/4816-proxmox-fix-extended-status.yaml]: Done!

Comment (Thulium-Drake) [reply changelogs/fragments/4816-proxmox-fix-extended-status.yaml]: Is there a way that I can skip the CI pipeline when making trivial typo fixes like these? :-)

**Comment by ilijamt** [comment]
> That should about cover it! @ilijamt can you have a look-see?

@Thulium-Drake Ran it on my local proxmox cluster. All looks good. 

```
plugin: community.general.proxmox
url: https://pve:8006
user: "ansible@pve"
password: "password"
validate_certs: yes
cache: no
want_facts: yes
want_proxmox_nodes_ansible_host: yes
qemu_extended_statuses: yes
keyed_groups:
  - key: proxmox_tags_parsed
    separator: ""
    prefix: ""
  - key: proxmox_ostype
    separator: "_"
    prefix: "proxmox_ostype"                                                                                                                                                                                                             ```

```
$ ansible-inventory -i my.proxmox.yml --graph
@all:
  |--@proxmox_all_lxc:
  |  |--k8s-lb
  |--@proxmox_all_paused:
  |  |--k8s-n01
  |--@proxmox_all_prelaunch:
  |--@proxmox_all_qemu:
  |  |--k8s-n01
  |  |--k8s-n02
  |  |--k8s-n03
  |--@proxmox_all_running:
  |  |--k8s-lb
  |  |--k8s-n02
  |  |--k8s-n03
  |--@proxmox_all_stopped:
  |--@proxmox_nodes:
  |  |--pve
  |--@proxmox_ostype_alpine:
  |  |--k8s-lb
  |--@proxmox_ostype_l26:
  |  |--k8s-n01
  |  |--k8s-n02
  |  |--k8s-n03
  |--@proxmox_pool_k8s:
  |  |--k8s-lb
  |--@proxmox_pve_lxc:
  |  |--k8s-lb
  |--@proxmox_pve_qemu:
  |  |--k8s-n01
  |  |--k8s-n02
  |  |--k8s-n03
  |--@ungrouped:                                                                                                                                                                                                                        
```

**Comment by ilijamt** [comment]
LGTM

File: changelogs/fragments/4816-proxmox-fix-extended-status.yaml (hunk 3 above)
Comment (felixfontein) [reply changelogs/fragments/4816-proxmox-fix-extended-status.yaml]: Unfortunately no.

**Comment by felixfontein** [comment]
@Thulium-Drake thanks for fixing this!
@ilijamt thanks for reviewing and testing!

**Comment by Thulium-Drake** [comment]
@felixfontein any chance this can be released before 5.2.0? As the plugin is borked now if the user doesn't have qemu VMs on their proxmox system :-)

Unless 5.2.0 is around the corner, that's also fine!

**Pull request merged**